### PR TITLE
Feat/task 001 implementar autenticacao com senha

### PR DESCRIPTION
**Pull request by victorgmrqs**
## Summary by Sourcery

Implement password-based authentication by adding login, register, and logout API methods with cookie support and token fallback, update the login form and hooks to handle passwords, enhance context logout to clear tokens and call the backend, and update tests accordingly

New Features:
- Introduce API authentication endpoints (login, register, logout) supporting HttpOnly cookies and fallback JWT storage
- Implement password-based login form with validation and integrate it into the useLogin hook

Enhancements:
- Add credentials: 'include' to fetch calls to enable cookie support
- Update UserContext logout to invoke backend logout and clear auth tokens
- Refactor useLogin to call the new auth.login API and handle user context update

Tests:
- Add tests for password field rendering, empty-password validation, and correct login mutation arguments
- Mock and verify api.auth.logout behavior in UserContext tests

Chores:
- Remove hardcoded test user list from login page

### CODE REVIEW

File: src/contexts/UserContext.tsx [hunk 1]
```diff
@@ -41,12 +42,20 @@ export const UserProvider = ({ children }: { children: ReactNode }) => {
     }
   };
 
-  const logout = () => {
+  const logout = async () => {
```
[the remark this answers is not in the document]
Comment (victorgmrqs) [reply src/contexts/UserContext.tsx:45]: @cursor, can you take a look here?
